src/main/java/org/owasp/esapi/reference/validation/HTMLValidationRule.java

```diff
@@ -97,8 +97,33 @@ public String sanitize( String context, String input ) {
 		return safe;
 	}
 
+    /**
+     * This admittedly is an UGLY hack to ensure that issue 509 and it's corresponding
+     * fix in PR #510 does not break existing developer's existing code. Full
+     * details are described in GitHub issue 521.
+     */
+    private boolean ignoreIssue509Fix() {
+        boolean ignore = true;
```

Comment:
I think this should be false by default and throw the exception in the implementation?

Reply:
My rationale for making this false if because I wanted to ensure backwards compatibility (i.e., not breaking someone's production code) if the updated their ESAPI jar to 2.2.1.0 (or whatever we name the next release) but fail to update their production ESAPI.properties file, which I think is probably a pretty common thing to do. E.g., they may have ESAPI.MasterKey / ESAPI.MasterSalt set or have done other extensive customization to it so that a wholesale copying of the updated ESAPI.properties file (which you can't download from Maven Central so is a bit harder to get anyway) so they figure it is safer to leave as-is, trusting that won't break anything. And generally they would be right. So it they take that approach, at best they may look for new properties to add in, but they very well could miss one, especially given it's dubious name of "Validator.ValidationRule.getValid.ignore509Fix". Also, I'm still not 100% convinced that throwing an exception is the correct behavior. We have no way of knowing if the Javadoc just wasn't wrong and since AntiSamy still ends up cleanses these cases, not throwing an exception here isn't really introducing any security risk that I can see here.